src/common/reactNativeProjectHelper.ts

```diff
@@ -19,22 +18,44 @@ export class ReactNativeProjectHelper {
      * {operation} - a function that performs the expected operation
      */
     public isReactNativeProject(): Q.Promise<boolean> {
-        let currentPackage = new Package(this.workspaceRoot);
-        return currentPackage.dependencies().then(dependencies => {
-            return !!(dependencies && dependencies["react-native"]);
-        }).catch((err: Error) => {
-            return Q.resolve(false);
+        let deferred = Q.defer<any>();
+
+        exec("react-native -v", {
```

Comment:
The logic for executing processes should be centralized in `commandExecutor.ts`. Please refactor this to have a method that runs this react-native command and returns the version number (or lack of version, for non-react-native projects), so we can analyzing the meaning of that version in these two methods (and possibly other endpoints). You can use some logic similar to how we launch the packager (https://github.com/Microsoft/vscode-react-native/blob/master/src/common/commandExecutor.ts#L60)